### PR TITLE
Allow negative values in charts

### LINKED ISSUE #1959: Charts should accept negative numbers
In the _Revenue_ report, we might need to display negative numbers in some circumstances, so the `<Chart>` component needs to be able to handle that. Extensions might want to display negative numbers in charts too.

### PR DESCRIPTION
**Pull request by Aljullu**
Fixes #1959.

Adds the ability to display negative numbers to `<Chart>` component.

### Screenshots
_All negative:_
image
_Positive and negative values:_
image
_Bar charts:_
image

### Detailed test instructions:
Currently all our charts only display positive values, so you need to overwrite `chartData` before rendering them.

In https://github.com/woocommerce/woocommerce-admin/blob/master/client/analytics/components/report-chart/index.js#L125, add these lines:
```ES6
for ( let i = 0; i < chartData.length; i++ ) {
	chartData[ i ].primary.value = ( Math.random() - 0.5 ) * 100;
	chartData[ i ].secondary.value = ( Math.random() - 0.5 ) * 100;
}
```
- Go to any report and verify charts look good.
- Switch between line and bar chart.
- Try modifying the code snippet above to force other data combinations:

**Only negative values:**
```ES6
for ( let i = 0; i < chartData.length; i++ ) {
	chartData[ i ].primary.value = Math.random() * -100;
	chartData[ i ].secondary.value = Math.random() * -100;
}
```
  **All 0's:**
```ES6
for ( let i = 0; i < chartData.length; i++ ) {
	chartData[ i ].primary.value = 0;
	chartData[ i ].secondary.value = 0;
}
```
  **Beautiful data going up (used for the screenshots above) :chart_with_upwards_trend: :**
```ES6
for ( let i = 0; i < chartData.length; i++ ) {
	chartData[ i ].primary.value = ( Math.random() - 0.7 + i / 30 ) * 100;
	chartData[ i ].secondary.value = ( Math.random() - 0.7 + i / 30 ) * 100;
}
```

### CODE REVIEW

**Review by emdashcodes — APPROVED** [review]
Really nice work here @Aljullu! This is working great.

I think this still needs a `CHANGELOG` entry: https://github.com/woocommerce/woocommerce-admin/blob/ac4354375c9856d7ae8decb043620509e28c944a/packages/components/CHANGELOG.md -- but otherwise, pre-approving.

**Comment by Aljullu** [comment]
@justinshreve absolutely! Added in 3577d3688af69f11e187579ed4ee2ecf801fe1fe.

**Comment by LevinMedia** [comment]
Late to the party but this is a thing of beauty. Thanks @Aljullu !

File: packages/components/src/chart/d3chart/utils/test/axis.js [hunk 1]
```diff
@@ -240,19 +240,57 @@ describe( 'compareStrings', () => {
...
 describe( 'getYGrids', () => {
-	it( 'returns a single 0 when yMax is 0', () => {
-		expect( getYGrids( 0 ) ).toEqual( [ 0 ] );
+	it( 'returns a single 0 when yMax and yMin are 0', () => {
+		expect( getYGrids( 0, 0, 0 ) ).toEqual( [ 0 ] );
 	} );
 
-	it( 'returns decimal values when yMax is <= 1', () => {
-		expect( getYGrids( 1 ) ).toEqual( [ 0, 0.3333333333333333, 0.6666666666666666, 1 ] );
+	describe( 'positive charts', () => {
```
Comment (timmyc) [inline packages/components/src/chart/d3chart/utils/test/axis.js]: @Aljullu these tests, 😮 , 👏

**Pull request merged**